cmd/distribute/distribute.go

```diff
@@ -101,8 +139,31 @@ func distributeReward() error {
 				return err
 			}
 		}
+
+		// Append bucket information to the list. For using the api to distribute the reward to the bucket.
+		for i := range toAccountAddrList {
+			allToBucketAddressList =
+				append(allToBucketAddressList,
+					toAccountAddrList[i])
+			allToBucketIDList =
+				append(allToBucketIDList,
+					toBucketIDList[i])
+			allToBucketAmountList =
+				append(allToBucketAmountList,
+					toBucketAmountList[i])
+			allToBucketDelegateNameList =
+				append(allToBucketDelegateNameList, dist.DelegateName)
+		}
 	}
-	return commitDistributions(c, endEpoch, delegateNames)
+	if err := commitDistributions(c, endEpoch, delegateNames); err != nil {
+		return err
+	}
+
+	if _, err := model.CreateJob(tip, endEpoch, allToBucketDelegateNameList, allToBucketAddressList, allToBucketIDList, allToBucketAmountList); err != nil {
```

Comment:
CreateJob should before commitDistributions, because we can't revert chain action when database error.